src/conventional/amdar_bufr2ioda.py

```diff
@@ -0,0 +1,506 @@
+            logging.warning("Caution, unable to find requested BUFR variable: " + variable)
+            vals[variable] = np.full(len(year), float_missing_value, dtype=np.float32)
+
+    # Be done with this BUFR message.
+    codes_release(bufr)
+
+    # For any of the MetaData elements that were totally lacking, fill entire vectory with missing.
+    for k, v in metaDataKeyList.items():
+        if not any(meta_data[k]):
+            meta_data[k] = assign_missing_meta(k, year)
+
+    # Need to transform some variables to others (wind speed/direction to components for example).
+    uwnd = np.full(len(year), float_missing_value)
+    vwnd = np.full(len(year), float_missing_value)
+    n = 0
+    for wind_direction in vals['windDirection']:
+        if (wind_direction != float_missing_value and vals['windSpeed'][n] != float_missing_value):
+            uwnd[n], vwnd[n] = met_utils.dir_speed_2_uv(wind_direction, vals['windSpeed'][n])
+        n+=1
+
+    spfh = np.full(len(year), float_missing_value)
+    n = 0
+    for mixing_ratio in vals['mixingRatio']:
+        if (mixing_ratio>0 and mixing_ratio<25.E-3):
+            spfh[n] = mixing_ratio / (1.0 + mixing_ratio)
+        n+=1
+
+    # Move everything into the final data dictionary, including metadata.
+    data['eastward_wind'] = np.append(data['eastward_wind'], uwnd)
+    data['northward_wind'] = np.append(data['northward_wind'], vwnd)
+    data['specific_humidity'] = np.append(data['specific_humidity'], spfh)
+    data['air_temperature'] = np.append(data['air_temperature'], vals['airTemperature'])
+    for key in meta_keys:
+        data[key] = np.append(data[key], meta_data[key])
+
+    logging.info("BUFR message number: " + str(count[0]))
+    logging.info("number of observations so far: " + str(count[1]))
+    logging.info("number of invalid or useless observations: " + str(count[2]))
+    return data, count, start_pos
+
+
+if __name__ == "__main__":
+
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description=(
+            'Read aircraft (AMDAR) BUFR file and convert into IODA output file')
+    )
+
+    required = parser.add_argument_group(title='required arguments')
+    required.add_argument('-i', '--input-files', nargs='+', dest='file_names',
+                          action='store', default=None, required=True,
+                          help='input files')
+    required.add_argument('-o', '--output-file', dest='output_file',
+                          action='store', default=None, required=True,
+                          help='output file')
+
+    optional = parser.add_argument_group(title='optional arguments')
+    optional.add_argument('--debug', action=argparse.BooleanOptionalAction,
+                          help='enable debug messages')
+    optional.add_argument('--verbose', action=argparse.BooleanOptionalAction,
+                          help='enable verbose debug messages')
+
+    args = parser.parse_args()
```

Comment:
```suggestion
    args = parser.parse_args()
    args.output_file = os.path.abspath(args.output_file)
```
I suggest this because IODA will complain if one is running to generate a file locally (not specifying a full path, e.g., path+filename). For example, running like this will crash:
```
amdar_bufr2ioda.py -i input.bufr -o output.nc4
```
with the following message:
```
OSError: output.nc4 does not specify a valid path to a file
```
While a `./output.nc4` would need to be passed instead. With the above suggestion that wouldn't be needed.

Reply:
@fabiolrdiniz, FYI the issue with the output file specification has been fixed in: https://github.com/JCSDA-internal/ioda/pull/601.

Reply:
Thank you very much, @srherbener! Sorry, I wasn't aware. @gthompsnJCSDA, I'll mark it as resolved.